packages/transformer-variant-group/src/index.ts

```diff
@@ -26,7 +26,7 @@ export default function transformerVariantGroup(
     name: '@unocss/transformer-variant-group',
     enforce: 'pre',
     transform(s) {
-      const result = parseVariantGroup(s, options.separators)
+      const result = parseVariantGroup(s, options.separators || [...defaultVariantGroupSeparators, '!', ''])
```

Comment:
I am not so sure if we should treat it this special. `!text-sm` is a syntax introduced by `preset-mini`, while `@unocss/transformer-variant-group` should be rule-agnostic as possible.

Reply:
I'm thinking if we could use a vscode plugin to provide some conversion functionality to reduce the work of unocss at runtime. Maybe I can implement `!(text-red bg-red)` into `!text-red !bg-red` in [`unot`](https://github.com/Simon-He95/unot)

Reply:
> I am not so sure if we should treat it this special. `!text-sm` is a syntax introduced by `preset-mini`, while `@unocss/transformer-variant-group` should be rule-agnostic as possible.

I'm thinking if we could use a vscode plugin to provide some conversion functionality to reduce the work of unocss at runtime. Maybe I can implement !(text-red bg-red) into !text-red !bg-red in [unot](https://github.com/Simon-He95/unot)